aten/src/ATen/native/cuda/MultinomialKernel.cu

```diff
@@ -71,7 +67,7 @@ void renormRows(Tensor& t) {
 
   AT_DISPATCH_FLOATING_TYPES_AND_HALF(t.scalar_type(), "renormRows_cuda", [&] {
     renormRowsL1<scalar_t>
-        <<<grid, block, block.x * sizeof(scalar_t),
+        <<<grid, block, (block.x / C10_WARP_SIZE) * sizeof(scalar_t),
```

Comment:
BlockReduceSum is calling `warp_shfl_down` with full mask, which requires all threads to participate. So you need to ensure that block.x is a multiple of WARP_SIZE (and make sure that added threads don't attempt to read values)